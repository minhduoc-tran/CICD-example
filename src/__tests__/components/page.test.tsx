import { render, screen } from "../utils";
import Home from "../../app/page";

describe("Home Page", () => {
  describe("Rendering and Functionality", () => {
    it("should render the main heading", () => {
      render(<Home />);

      const heading = screen.getByText("Next.js CI/CD with Docker");
      expect(heading).toBeInTheDocument();
      expect(heading.tagName).toBe("H1");
    });

    it("should render the description paragraph", () => {
      render(<Home />);

      const description = screen.getByText(
        /This is a simple example of a Next.js application/
      );
      expect(description).toBeInTheDocument();
      expect(description.tagName).toBe("P");
    });

    it("should have correct CSS classes for styling", () => {
      render(<Home />);

      const heading = screen.getByText("Next.js CI/CD with Docker");
      const container = heading.parentElement;
      expect(container).toHaveClass(
        "flex",
        "min-h-screen",
        "items-center",
        "justify-center"
      );
    });

    it("should have proper semantic structure", () => {
      render(<Home />, { skipProviders: true });

      // Use getByText instead of getByRole to avoid CSS-related issues
      const heading = screen.getByText("Next.js CI/CD with Docker");
      expect(heading.tagName).toBe("H1");

      // Check that the description is properly associated
      const description = screen.getByText(/This is a simple example/);
      expect(description).toBeInTheDocument();
      expect(description.tagName).toBe("P");
    });

    it("should have accessible text content", () => {
      render(<Home />, { skipProviders: true });

      // Test accessibility by checking for heading element
      const heading = screen.getByText("Next.js CI/CD with Docker");
      expect(heading).toBeInTheDocument();
      expect(heading.tagName).toBe("H1");

      // Ensure text is readable and not empty
      expect(heading.textContent?.trim()).toBeTruthy();
      expect(heading.textContent?.length).toBeGreaterThan(0);
    });
  });

  describe("Provider Integration", () => {
    it("should render with providers (testing custom render)", () => {
      render(<Home />, {
        providerProps: {
          authProps: { isAuthenticated: true },
        },
      });

      // Verify the component renders correctly with providers
      expect(screen.getByText("Next.js CI/CD with Docker")).toBeInTheDocument();
    });

    it("should render without providers when skipProviders is true", () => {
      render(<Home />, { skipProviders: true });

      // Verify the component renders correctly without providers
      expect(screen.getByText("Next.js CI/CD with Docker")).toBeInTheDocument();
    });
  });

  describe("Snapshot Tests", () => {
    it("should match snapshot", () => {
      const { container } = render(<Home />, { skipProviders: true });
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should match snapshot with providers", () => {
      const { container } = render(<Home />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
