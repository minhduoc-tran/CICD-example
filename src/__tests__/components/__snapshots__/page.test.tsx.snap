// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home Page Snapshot Tests should match snapshot 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-zinc-50 font-sans dark:bg-black"
>
  <h1
    class="text-4xl font-bold text-zinc-800 dark:text-zinc-200"
  >
    Next.js CI/CD with Docker
  </h1>
  <p
    class="mt-4 text-lg text-zinc-600 dark:text-zinc-400"
  >
    This is a simple example of a Next.js application with CI/CD using Docker.
  </p>
</div>
`;

exports[`Home Page Snapshot Tests should match snapshot with providers 1`] = `
<div
  data-testid="mock-theme-provider"
>
  <div
    data-testid="mock-router-provider"
  >
    <div
      data-authenticated="false"
      data-testid="mock-auth-provider"
    >
      <div
        class="flex min-h-screen items-center justify-center bg-zinc-50 font-sans dark:bg-black"
      >
        <h1
          class="text-4xl font-bold text-zinc-800 dark:text-zinc-200"
        >
          Next.js CI/CD with Docker
        </h1>
        <p
          class="mt-4 text-lg text-zinc-600 dark:text-zinc-400"
        >
          This is a simple example of a Next.js application with CI/CD using Docker.
        </p>
      </div>
    </div>
  </div>
</div>
`;
